### ISSUE TITLE
[BUG] Issue with "Go to Implementations" feature opening decompiled file instead of source code

### ISSUE THREAD

**Issue by nphamvn** [microsoft/vscode-dotnettools#135]
### Describe the Issue

In my solution, I have two projects: a Web project (Web application) and a Domain project (Class library). The Web project references the Domain project. However, when I attempt to use the "Go to implementations" feature in the Web project to navigate to the implementation, it opens a temporary decompiled file instead of the source code in the Domain project.

**Comment by Krohitkrishna12** [comment]
I'm having the same issue on my machine (Debian Linux) as well. The references & go to definition work until i close the project or folder and from then onward it takes me to de-compiled metadata instead of actually going to the project. If i disable the C# Dev kit the references work as expected. So anyone having same issue seems like disabling the extension is the  only option to fix this until the issue is fixed.

**Comment by jeangatto** [comment]
I'm having the same problem here; I've tried it on 3 different repositories.
I cleaned up the temporary files, the bin and obj folder and still has this bug.

**Comment by tmitchel2** [comment]
I am having this problem now (Mac environment), my installation seems to have degraded over time somehow and even in a small test project I am now consistently getting this problem.

A little more detail from what i can see.. if you hover over a Type referenced from another project you see this info in the popover
```
Class1.Class1()
    /Users/blah/blah/csharptest.csproj ($net7.0) - Available
    csharptest - Not Available ⚠

You can use the navigation bar to switch contexts.
```
Which makes me think C# Dev Kit is not able to match up projects correctly ??

**Comment by lucky51** [comment]
I am having  the same issue (windows 11 env).

**Comment by tmitchel2** [comment]
I've just realised that my issues started when a new version v0.1.103 of C# Dev Kit was released, I have now manually gone back to v0.1.83 and the situation is a lot better.

**Comment by jasonmalinowski** [comment]
> I am having this problem now (Mac environment), my installation seems to have degraded over time somehow and even in a small test project I am now consistently getting this problem.
> 
> A little more detail from what i can see.. if you hover over a Type referenced from another project you see this info in the popover
> 
> ```
> Class1.Class1()
>     /Users/blah/blah/csharptest.csproj ($net7.0) - Available
>     csharptest - Not Available ⚠
> 
> You can use the navigation bar to switch contexts.
> ```
> 
> Which makes me think C# Dev Kit is not able to match up projects correctly ??

@tmitchel2 We expect that bug to be fixed not by the C# Dev Kit extension but a fix to the base C# extension that's installed as a dependency of Dev Kit. As long as you're on 2.0.212 (which we shipped yesterday) we expect that to be fine.
